### PR TITLE
feat: Respect new Eclipse Che Dashboard  deployment

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### What does this PR do?
This PR propagate Dashboard Deployment startup, export logs, stop on server:stop

Reff issue: https://github.com/eclipse/che/issues/19605

### Screenshot/screencast of this PR

image

### What issues does this PR fix or reference?

N/A 

### How to test this PR?

N/A 

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by flacatus** [comment]
Depends on https://github.com/eclipse-che/che-operator/pull/805

File: src/tasks/che.ts [hunk 1]
```diff
@@ -37,7 +37,7 @@ export class CheTasks {
   cheDeploymentName: string
 
   dashboardDeploymentName = 'che-dashboard'
-  dashboardSelector = 'app=che,component=che-dashboard'
+  dashboardSelector = 'app=che,component,component=che-dashboard'
```
Comment (tolusha) [inline src/tasks/che.ts:40]: Typo ?

File: src/tasks/che.ts [hunk 2]
```diff
@@ -84,17 +84,22 @@ export class CheTasks {
...
       {
-        title: 'Plugin registry pod bootstrap',
+        title: 'Plug-in Registry pod bootstrap',
         enabled: ctx => ctx.isPluginRegistryDeployed && !ctx.isPluginRegistryReady,
         task: () => this.kubeTasks.podStartTasks(this.pluginRegistrySelector, this.cheNamespace)
       },
       {
-        title: 'Eclipse Che pod bootstrap',
+        title: 'Eclipse Che Dashboard pod bootstrap',
+        enabled: ctx => ctx.isDashboardDeployed && !ctx.isDashboardReady,
```
Comment (tolusha) [inline src/tasks/che.ts:98]: I don't see if `ctx.isDashboardReady` is set somewhere

**Comment by flacatus** [comment]
@tolusha The tests will still fail because will detect `'RELATED_IMAGE_dashboard' environment in che-operator deployment in nightly version

File: src/tasks/che.ts (hunk 2 above)
Comment (flacatus) [reply src/tasks/che.ts:98]: Is here https://github.com/che-incubator/chectl/blob/dashboard_dpl/src/tasks/che.ts#L130-L136. There was already a implementation when dashboard deployment was added to helm

**Review by tolusha — APPROVED** [review]

**Pull request merged**